doc/developer/design/20230810_alter_connection.md

```diff
@@ -0,0 +1,64 @@
+# `ALTER CONNECTION`
+
+- Associated:
+    - https://github.com/MaterializeInc/materialize/issues/17620
+
+## Context
+
+A `CONNECTION` must be valid for a source to ingest data--however, in our
+current scheme, `CONNECTION`s are immutable so any change to the service hosting
+the source permanently wedges the source in Materialize. This is undesirable
+because the `DROP`/`CREATE` cycle for a source can be very painful for users
+because of rehydration.
+
+## Goals
+
+Allow users to modify a `CONNECTION` s/t they can repair the TCP connection
+between MZ and their services.
+
+## Non-Goals
+
+Allow users to meticulously control their sources' state changes.
+
+## Overview
+
+We can perform a process similar to `ALTER SOURCE` wherein we update some state
+in memory, re-plan something, update it in the catalog, and then communicate the
+updated state throughout the system.
+
+## Detailed description
+
+1. Parse `ALTER CONNECTION <name> AS (KAFKA|POSTGRES) SET <option> =? <val>`.
```

Comment:
```suggestion
1. Parse `ALTER CONNECTION <name> AS (KAFKA|POSTGRES) SET (<option> =? <val>)`.
```

Let me ponder this for a minute (or let's chat live on Slack). The `AS {KAFKA|POSTGRES}` is really too bad, ugh.

Reply:
```suggestion
1. Parse `ALTER CONNECTION <name> SET (<option> =? <val>)`.
```

On second thought, as discussed below, we think we can instead accept all options at parse time and restrict to the appropriate set during plan time.